state-chain/pallets/cf-swapping/src/tests.rs

```diff
@@ -2075,3 +2077,156 @@ fn can_swap_ccm_below_max_amount() {
 		assert_eq!(CollectedRejectedFunds::<Test>::get(from), 0);
 	});
 }
+
+fn swap_with_custom_broker_fee(
+	from: Asset,
+	to: Asset,
+	amount: AssetAmount,
+	broker_fee: BasisPoints,
+) {
+	const ALICE: u64 = 2_u64;
+
+	<Pallet<Test> as SwapDepositHandler>::schedule_swap_from_channel(
+		ForeignChainAddress::Eth([2; 20].into()),
+		Default::default(),
+		from,
+		to,
+		amount,
+		ForeignChainAddress::Eth([2; 20].into()),
+		ALICE,
+		broker_fee,
+		1,
+	);
+}
+
+#[test]
+fn swap_broker_fee_calculated_correctly() {
+	new_test_ext().execute_with(|| {
+		const ALICE: u64 = 2_u64;
+		let fees: [BasisPoints; 12] =
+			[1, 5, 10, 100, 200, 500, 1000, 1500, 2000, 5000, 7500, 10000];
+		const AMOUNT: AssetAmount = 100000;
+		const BASIS_POINTS_PER_MILLION: u32 = 100;
+		// calculate broker fees for each asset available
+		Asset::all().iter().for_each(|asset| {
+			let total_fees: u128 =
+				fees.iter().fold(0, |total_fees: u128, fee_bps: &BasisPoints| {
+					swap_with_custom_broker_fee(*asset, Asset::Usdc, AMOUNT, *fee_bps);
+					total_fees +
+						Permill::from_parts(*fee_bps as u32 * BASIS_POINTS_PER_MILLION) * AMOUNT
+				});
+			assert_eq!(EarnedBrokerFees::<Test>::get(ALICE, asset), total_fees);
+		});
+	});
+}
+
+#[test]
+fn swap_broker_fee_cannot_exceed_amount() {
+	new_test_ext().execute_with(|| {
+		const ALICE: u64 = 2_u64;
+		swap_with_custom_broker_fee(Asset::Usdc, Asset::Flip, 100, 15000);
+		assert_eq!(EarnedBrokerFees::<Test>::get(ALICE, cf_primitives::Asset::Usdc), 100);
+	});
+}
+
+fn swap_scheduled_event_witnessed(
+	swap_id: u64,
+	source_asset: Asset,
+	deposit_amount: AssetAmount,
+	destination_asset: Asset,
+	broker_commission: Option<AssetAmount>,
```

Comment:
nit: we always pass in a `Some()` to this, so we can dedup the wrapping and just wrap it in some in here